### DISCUSSION TITLE
about my thoughts

### DISCUSSION THREAD

**Discussion by 1996-Chinese-Chen** [microsoft/aspire#2044] [Q&A]
Can you add a configuration center here, where other projects can read public or specific project configurations?
If possible, can I change the Transport or Application of IDuplexPipe so that I can read the information of each request and response?

**Comment by 1996-Chinese-Chen** [comment]
Has anyone seen it

**Reply by davidfowl** [reply]
I'm not 100% sure what you mean by configuration center, but maybe you mean a configuration API or something similar? We have no plans to build a customer configuration manager in aspire, but you can use one of the many configuration systems that exists with aspire.

**Reply by 1996-Chinese-Chen** [reply]
![image](https://github.com/dotnet/aspire/assets/75053585/2b3d5a91-7450-40d0-8548-59ec403c9fb7)
What I mean is to add a Configuration here, and then all Projects can read the configuration here

**Reply by davidfowl** [reply]
Nice, I like the idea, so this is a development only scenario right?

**Reply by 1996-Chinese-Chen** [reply]
I am going to get the deletion and delete the configuration through the GRPC, the configuration is stored in the Redis, and use the iconfiguration in the application to obtain the configuration from the Redis. Now there is a problem that I use IDISTRIBUTEDCACHE. The redisCache default is the storage of Hash. This is the storage. Can't change, at the same time I don't want to implement it myself, hahahahaha, do you have any good suggestions, because adding a cache in the Apphost also requires the application name. I don't have any good ideas at present, unless I realize IdistributedCache, use Stackexchange inside .Rdis

**Comment by 1996-Chinese-Chen** [comment]
Hello, regarding the configuration aspect, this time I found that WaterAsync has been introduced to observe changes in Resources. However, how to use WatchAsync to observe changes in Resources in other projects? The Resource Notification Service depends on Aspire I can obtain the Hosting DLL from the apphost project, but it cannot be obtained from other web projects, let alone monitoring changes in Resources

**Reply by davidfowl** [reply]
What do you mean other web projects? This only works in process in the app host. You can build libraries that listen to notifications while the app host is running.

**Reply by 1996-Chinese-Chen** [reply]
I don't know if you remember the idea of the configuration center I mentioned last time. I feel that resource can be realized. You can set up the configuration in apphost, and then monitor the changes through the Resource Notification Service in the web project to get the latest information.

**Reply by davidfowl** [reply]
The web project (assuming this project is a normal orchestrated project) cannot get notifications from the app host, at least, not using this mechanism.

**Reply by 1996-Chinese-Chen** [reply]
Then the host project configuration change notice cannot be realized.

**Reply by davidfowl** [reply]
Or design some other way to communicate changes to the application.

**Reply by 1996-Chinese-Chen** [reply]
Last time I saw the Resource Notification Service, I thought it was okay. Haha, I tried it and it didn't work. I'm looking forward to the new design.

**Reply by davidfowl** [reply]
Here's a doc on the resource model: https://gist.github.com/davidfowl/b408af870d4b5b54a28bf18bffa127e1
